starter-code/src/viking.js

```diff
@@ -1,11 +1,110 @@
 // Soldier
-function Soldier () {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+
+Soldier.prototype.attack = function() {
+  return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function(damage) {
+  this.health = this.health - damage;
+}
 
 // Viking
-function Viking () {}
+function Viking(name, health, strength) {
+  Soldier.call(this, health, strength);
+  this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+Viking.prototype.receiveDamage = function(damage) {
+  this.health = this.health - damage;
+  if (this.health > 0) {
+    return this.name + ' has received ' + damage + ' points of damage';
+  } else {
+    return this.name + ' has died in act of combat';
+  }
+}
+
+Viking.prototype.battleCry = function() {
+  return 'Odin Owns You All!';
+}
 
 // Saxon
-function Saxon () {}
+function Saxon(health, strength) {
+  Soldier.call(this, health, strength);
+  this.attack();
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+
+Saxon.prototype.receiveDamage = function(damage) {
+  this.health = this.health - damage;
+  if (this.health > 0) {
+    return 'A Saxon has received ' + damage + ' points of damage';
+  } else {
+    return 'A Saxon has died in combat';
+  }
+}
 
 // War
-function War () {}
+function War() {
+  this.vikingArmy = [];
+  this.saxonArmy = [];
+}
+
+War.prototype.addViking = function(Viking) {
+  this.vikingArmy.push(Viking);
+};
+
+War.prototype.addSaxon = function(Saxon) {
+  this.saxonArmy.push(Saxon);
+};
+
+War.prototype.vikingAttack = function() {
+  var randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+  var randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+  var saxonRecieveDamage = this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].strength);
+
+  if (this.saxonArmy[randomSaxon].health <= 0) {
+    this.saxonArmy.pop(this.saxonArmy[randomSaxon]);
```

Comment:
The pop method doesn't accept an argument, always remove the LAST element of the array. If you want to remove one specifc element in an array... ¡¡¡SEARCH IN GOOGLE!!!

Ok, here you have a hint, but don't get used: https://stackoverflow.com/questions/5767325/how-do-i-remove-a-particular-element-from-an-array-in-javascript